decidim-initiatives/app/permissions/decidim/initiatives/permissions.rb

```diff
@@ -131,6 +137,13 @@ def unvote_initiative?
         toggle_allow(can_unvote)
       end
 
+      def initiative_attachment?
+        return unless permission_action.action == :add_attachment &&
+                      permission_action.subject == :initiative
+
+        toggle_allow(initiative_type.attachments_enabled?)
```

Comment:
```suggestion
        toggle_allow(initiative_type&.attachments_enabled?)
```
As `initiative_type` might be nil, should we be extra careful here?

Reply:
Hi @microstudi, thanks!
I'm not sure we need that, at the point we check that attachments permission the initiative_type is required. It can be nil only in the first step of the creation wizard and we check that on the third/fourth and on edit.
Do you think it is worth adding it just in case a future change may need it?

Reply:
Well, I'd put it, unless we are sure that initiative_type will never be nil. In which case:
```ruby
      def initiative_type
        @initiative_type ||= context.fetch(:initiative_type, nil)
      end
```
Shouldn't return a default nil value. Just for coherence. But you are right, it will probably never be a problem in real life.

Reply:
Ah, I see your point, you're right. That fetch is wrong. There should always be an initiative_type at that point. If there is not, then we have an error, there's no point in having a nil default. Thanks for pointing it out!